src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+            "LVM ERROR: Failed to check available disk space.");
+    }
+}
+
+void LVM::verifyAvailablePartitions()
+{
+    std::list<std::string> output;
+    const std::string checkPartitionsCommand
+        = "lsblk --noheadings --output MOUNTPOINT";
+    int exitCode = cloyster::runCommand(checkPartitionsCommand, output, false);
+
+    if (exitCode == 0) {
+        for (const auto& line : output) {
+            if (line == "/var") {
+                m_hasVarPartition = true;
+            } else if (line == "/opt") {
+                m_hasOptPartition = true;
+            } else if (line == "/home") {
+                m_hasHomePartition = true;
+            }
+        }
+
+        if (m_hasHomePartition) {
+            LOG_TRACE(
+                "LVM: /home is in a separate partition. We can proceed with "
+                "LVM snapshot.");
+        } else {
+            LOG_WARN("LVM: /home is not a separate partition. This may lead "
+                     "to issues when rolling back the snapshot.");
+        }
+    } else {
+        throw std::runtime_error(
+            "LVM ERROR: Failed to list available partitions.");
+    }
+}
+void LVM::checkLVMAvailability()
+{
+    LOG_INFO("LVM: Begin of availability check.")
+    verifyBootIsNotLVM();
+    checkUEFIMode();
+    checkLVMEnabled();
+    checkThinProvisioning();
+    checkEnoughDiskSpaceAvailable();
+    verifyAvailablePartitions();
+    LOG_INFO("LVM: End of availability check.")
+}
+
+bool LVM::snapshotExists(const std::string& snapshotName)
+{
+    std::list<std::string> output;
+
+    const std::string checkSnapshotCommand
+        = fmt::format("lvs --noheadings -o lv_name {}/{}",
+            m_snapshotVolumeGroup, snapshotName);
+    int exitCode = cloyster::runCommand(checkSnapshotCommand, output, false);
+
+    // If the command succeeded (exitCode 0) and output is not empty, snapshot
+    // exists
+    if (exitCode == 0 && !output.empty()) {
+        std::string lvName = output.front();
+
+        // Trim leading/trailing whitespaces
+        lvName.erase(lvName.begin(),
+            std::find_if(lvName.begin(), lvName.end(),
+                [](unsigned char ch) { return !std::isspace(ch); }));
+        lvName.erase(std::find_if(lvName.rbegin(), lvName.rend(),
+                         [](unsigned char ch) { return !std::isspace(ch); })
+                         .base(),
+            lvName.end());
+
+        // Check if the name matches the expected snapshot name
+        if (lvName == snapshotName) {
+            LOG_INFO("LVM: Snapshot {} exists.", snapshotName);
+            return true;
+        } else {
+            LOG_INFO("LVM: Snapshot found but name mismatch: expected '{}', "
+                     "got '{}'",
+                snapshotName, lvName);
+        }
+    }
+
+    // If the command failed or the snapshot was not found
+    LOG_INFO("LVM: Snapshot {} does not exist.", snapshotName);
+    return false;
+}
+
+void LVM::verifyBootIsNotLVM()
```

Comment:
By default on EL, boot is not on LVM due to compatibility issues. However it can be inside LVM, and if it is, it should not abort because of that.

So it should not be an issue.

Reply:
I kindly ask that you read your list again.

Reply:
Also this method should return a boolean, the name if verify. So it is expected to return.